syntax = "proto3";

package taskmanager;

// TaskManager service handles task-related operations
service TaskManager {
  rpc GetTask (TaskRequest) returns (TaskResponse);
  rpc AddTask (Task) returns (TaskResponse);
  rpc ListTasks (Empty) returns (TaskList);
  rpc DeleteTask (TaskRequest) returns (DeleteResponse);
  rpc UpdateTask (UpdateTaskRequest) returns (TaskResponse);
}

message TaskRequest {
  string taskId = 1;
}

message Task {
  string id = 1;
  string title = 2;
  string description = 3;
  string status = 4;
  string createdAt = 5;
}

message TaskResponse {
  Task task = 1;
  string message = 2;
}

message Empty {
}

message TaskList {
  repeated Task tasks = 1;
}

message DeleteResponse {
  bool success = 1;
  string message = 2;
}

message UpdateTaskRequest {
  string taskId = 1;
  Task task = 2;
}
